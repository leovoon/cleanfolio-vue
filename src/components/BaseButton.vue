<template>
  <button
    class="border border-2 text-center px-4 py-2 transition-colors ease-out cursor-pointer text-blue-primary border-blue-primary dark:(border-purple-dark text-purple-dark hover:text-purple-light hover:border-purple-light)"
  >
    {{ title }}
  </button>
</template>

<script lang="ts" setup>
defineProps({
  title: {
    type: String,
    default: 'Button',
  },
})
</script>
