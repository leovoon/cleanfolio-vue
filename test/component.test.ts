import { mount } from '@vue/test-utils'
import BaseButton from '@/components/BaseButton.vue'
describe('BaseButton.vue', () => {
  it('should render', () => {
    const wrapper = mount(BaseButton, { props: { title: 'any text' } })
    expect(wrapper.text()).toContain('any text')
    expect(wrapper.html()).toMatchSnapshot()
  })
})

//   it('should be interactive', async() => {
//     const wrapper = mount(Counter, { props: { initial: 0 } })
//     expect(wrapper.text()).toContain('0')

//     expect(wrapper.find('.inc').exists()).toBe(true)

//     await wrapper.get('button').trigger('click')

//     expect(wrapper.text()).toContain('1')
//   })
// })
