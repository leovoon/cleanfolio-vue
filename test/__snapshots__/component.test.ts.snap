// Vitest Snapshot v1

exports[`BaseButton.vue > should render 1`] = `"<button class=\\"border border-2 text-center px-4 py-2 text-blue-primary border-blue-primary dark:border-purple-dark dark:text-purple-dark\\">any text</button>"`;
